read footprint from truck namespace instead of coordinator

ksatyaki opened the pull request:
This enables footprints to be read from robot specific namespaces,thus allowing heterogeneous robots (with different footprints), which the coordinator already supports. For backwards compatibility, the footprints should be systematically moved from coordination launch files to the robot's param files. This is done already in the tmule-based system, but needs to be done on the older launch-file-based system as well.

Suggested additional change: the coordinator should wait (with a timeout) for the parameters to appear. This avoids restarting the coordinator if it is started before the robots. This isn't a problem (probably) in the old launch-file-based system.

annamannucci commented:
Hi all!!

Thank you Chitt. It seems reasonable to me.

Since robots' footprints are mandatory, we can force the coordinator to
wait for them (timeout + kill the node in case of missed parameters for
example).
Note that launch files do not ensure an order in starting ros nodes,
synchronization should be forced via code.
I suggest to extend these changes to the maximum velocity, the maximum
acceleration and robot control period too to let parameters to be
customized for each robot.

Cheers,

Anna

Il giorno gio 6 giu 2019 alle ore 17:29 Chittaranjan Srinivas Swaminathan <
notifications@github.com> ha scritto:

> This enables footprints to be read from robot specific namespaces,thus
> allowing heterogeneous robots (with different footprints), which the
> coordinator already supports. For backwards compatibility, the footprints
> should be systematically moved from coordination launch files to the
> robot's param files. This is done already in the tmule-based system, but
> needs to be done on the older launch-file-based system as well.
>
> Suggested additional change: the coordinator should wait (with a timeout)
> for the parameters to appear. This avoids restarting the coordinator if it
> is started before the robots. This isn't a problem (probably) in the old
> launch-file-based system.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/FedericoPecora/coordination_oru_ros/pull/9
> Commit Summary
>
>    - read footprint from truck namespace instead of coordinator
>
> File Changes
>
>    - *M*
>    coordination_oru_ros/coordinator/src/main/java/se/oru/coordination/coordinator/ros_coordinator/generic/MainNode.java
>    <https://github.com/FedericoPecora/coordination_oru_ros/pull/9/files#diff-0>
>    (34)
>
> Patch Links:
>
>    - https://github.com/FedericoPecora/coordination_oru_ros/pull/9.patch
>    - https://github.com/FedericoPecora/coordination_oru_ros/pull/9.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FedericoPecora/coordination_oru_ros/pull/9?email_source=notifications&email_token=AESZFB2RAS4ON3HHOE4PL6TPZEUOZA5CNFSM4HVDFGQ2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4GYBXFZQ>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AESZFB3DVCJ26FTTQVXS3EDPZEUOZANCNFSM4HVDFGQQ>
> .
>

ksatyaki commented:
According to Anna's suggestions the max_vel and max_accel are read from the controller's params:
robot<robot_no>/execution/max_vel
robot<robot_no>/execution/max_acc

The following launch files were also fixed:
1. multiple_trucks.launch
2. multiple_trucks_aass_basement.launch
3. multiple_trucks_aass_basement_point_and_click.launch **(same file as above, deleted)**
4. multiple_trucks_aass_basement_point_and_click_iliad_smp.launch **(deprecated file removed. Luigi uses `iliad_metapackage`)**
5. multiple_trucks_arena*.launch **(old ijcai demo files, can easily be reincarnated from existing files or git-tags.)**
6. multiple_trucks_iliad_smp.launch **(deprecated, deleted)**
7. multiple_trucks_robotlab_point_and_click.launch **(Changed one robot to have a bigger footprint ;-) )**
8. multiple_trucks_robotlab_point_and_click_remote.launch **(should work, given the above works)**
9. robot_lab_demo_two_trucks_smp.launch **(deprecated, deleted)**
10. ncfm_demo_*.launch **(deprecated, deleted)**
11. multiple_trucks_aass_basement_3_rviz_viewpoints.launch **(configs don't exist, deleted)**
12. robot_lab_demo*.launch **(error loading urdf, deleted. I'm assuming this hasn't been used for a long time)**
13. empty_world_one_truck_manipulation_test.launch **(seems like another deprecated file)**

I added some color coded output since this greatly helps identifying errors should there be any. Please let me know if you dislike these. ;-)

Thanks
Chittaranjan

ksatyaki commented:
I am quite sure this can now be safely merged into master. I will leave it to @annamannucci or @FedericoPecora to do it.

ksatyaki commented:
Can this be merged?

FedericoPecora commented:
Done and deleted branch.

Best, F.

On Tue, Sep 10, 2019 at 2:05 PM Chittaranjan Srinivas Swaminathan <
notifications@github.com> wrote:

> Can this be merged?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FedericoPecora/coordination_oru_ros/pull/9?email_source=notifications&email_token=ACJBG3MANMOP5COYYXNHXFTQI6ER7A5CNFSM4HVDFGQ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6K3C3I#issuecomment-529903981>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACJBG3NNW4ECABJCKRUSPY3QI6ER7ANCNFSM4HVDFGQQ>
> .
>


-- 
--~--~---------~--~----~------------~-------~--~----------~--~--~---------~--~-----
Federico Pecora, PhD, Associate Professor
AASS Cognitive Robotic Systems Lab fpa@aass.oru.se
School of Science and Technology www.aass.oru.se/~fpa
Örebro University Tel: +46 19 30 3319
S-70182 Örebro, Sweden Fax: +46 19 30 3463
--~--~---------~--~----~------------~-------~--~----------~--~--~---------~--~-----
